src/messageschannel.hpp

```diff
@@ -75,6 +75,8 @@ class MatrixMessagesChannel : public Tp::BaseChannelTextType
     void sendDeliveryReport(Tp::DeliveryStatus tpDeliveryStatus, const QString &deliveryToken);
     void onPendingEventChanged(int pendingEventIndex);
     void onReadMarkerForUserMoved(Quotient::User* user, const QString &fromEventId, const QString &toEventId);
+    void onDisplayNameChanged(Quotient::Room *room, QString oldName);
```

Comment:
Please avoid strings copy by using `const QString &` as in the methods above.

Reply:
I'll probably do it this weekend, I managed to break my workflow by installing NixOS, and now I can't compile programs anymore with CMake. I already spent way to much time on trying to fix it, but I also have other stuff to do.

Reply:
You can do the change without a (compile) check. It will work, I bet you :)
1. Replace "`QString oldName`" by "`const QString &oldName`" here.
2. Replace "`QString oldName`" by "`const QString &oldName`" in .cpp.
3. `git add -u; git commit --amend --no-edit; git push -f`.
You're done.

Reply:
There you go. I was afraid it wouldn't compile since it's a slot which connects to a signal which has a regular QString, not a constant reference.